### PR TITLE
Add additional enum Test

### PR DESCRIPTION
**Pull request by emlyn-rice**
Protect background service from exceptions
Change expected state to be a list
Fix key issues when looking up feature keys
Added FeatureGateAttributeBase to allow for gates on controller actions.

### CODE REVIEW

File: libs/dotnet-sdk/README.md [hunk 1]
```diff
@@ -10,11 +10,13 @@ dotnet add package FeatureBoard.DotnetSdk
 ## Setup
 
 1) Create a Features model
+Note that FeatureKeyName can be used here to apply the correct feature key to a property, otherwize feature board will do it's best to convert the property to a kebab case key.
```
Comment (JakeGinnivan) [inline libs/dotnet-sdk/README.md]: otherwise

File: libs/dotnet-sdk/State/FeatureBoardState.cs [hunk 2]
```diff
@@ -9,50 +9,49 @@ public class FeatureBoardState : IFeatureBoardState
...
-  public async Task InitialiseState(Dictionary<string, FeatureConfiguration>? features, DateTimeOffset? lastModified, CancellationToken cancellationToken)
+  public async Task InitialiseState(List<FeatureConfiguration>? features, DateTimeOffset? lastModified, CancellationToken cancellationToken)
   {
     if (features is null && _externalState is not null)
     {
       features = await _externalState.GetState(cancellationToken);
     }
 
-    foreach (var featureConfiguration in features ?? new Dictionary<string, FeatureConfiguration>())
+    foreach (var feature in features ?? new List<FeatureConfiguration>())
```
Comment (JakeGinnivan) [inline libs/dotnet-sdk/State/FeatureBoardState.cs]: This won't remove items which have been removed from the passed in list?

Comment (JakeGinnivan) [reply libs/dotnet-sdk/State/FeatureBoardState.cs]: Possible this method needs to throw if it's already been initialised?

File: libs/dotnet-sdk/README.md (hunk 1 above)
Comment (emlyn-rice) [reply libs/dotnet-sdk/README.md]: Resolved

File: libs/dotnet-sdk/State/FeatureBoardState.cs (hunk 2 above)
Comment (emlyn-rice) [reply libs/dotnet-sdk/State/FeatureBoardState.cs]: Updated to thow

File: libs/dotnet-sdk/README.md (hunk 1 above)
Comment (emlyn-rice) [reply libs/dotnet-sdk/README.md]: fixed

File: libs/dotnet-sdk/State/FeatureBoardState.cs (hunk 2 above)
Comment (emlyn-rice) [reply libs/dotnet-sdk/State/FeatureBoardState.cs]: Fixed

**Pull request merged**